src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -874,53 +868,43 @@ public without sharing class BDI_MigrationMappingUtility {
 
             if (this.helpText != null) {
 
-                List<String> parts = splitHelpText(this.helpText);
-                if (parts != null) {
-                    this.dataImportObjectName = String.isNotBlank(parts[0]) ? parts[0] : null;
-                    this.targetFieldAPIName = String.isNotBlank(parts[1]) ? parts[1] : null;
-
-                    if (UTIL_Namespace.shouldAlignNamespace) {
-                        this.targetFieldAPIName =
-                            UTIL_Namespace.alignSchemaNSWithEnvironment(this.targetFieldAPIName);
-                    }
+                List<String> mappingValuesFromHelpText = retrieveObjectAndFieldApiNames(this.helpText);
+                if (mappingValuesFromHelpText != null) {
+                    this.dataImportObjectName = mappingValuesFromHelpText[0];
+                    this.targetFieldAPIName =
+                        UTIL_Namespace.shouldAlignNamespace ?
+                            UTIL_Namespace.alignSchemaNSWithEnvironment(mappingValuesFromHelpText[1]) :
+                            mappingValuesFromHelpText[1];
 
                     checkRequiredFields();
                 }
             }
         }
 
-        private List<String> splitHelpText(String helpText) {
-            String symbol;
-            Boolean hasSpecialChar = helpText.containsAny('$&+,:;=?@#|\'<>.-^*()%!_');
-            Boolean hasDataImportNames =
-                helpText.containsAny('Account1Account2AddressContact1Contact2HouseholdOpportunityDonationPaymentGAURole');
-
-            if (hasSpecialChar == true) {
-                String[] specialCharacters =
-                    new List<String>{'$', '&', '\\+', ',', ':', ';', '=', '\\?', '@',
-                                     '#', '\\|', '\'', '<', '>', '\\.', '-', '^', '\\*',
-                                     '\\(', '\\)', '%', '!', '_'};
-
-                for (String sc : specialCharacters) {
-                    List<String> parts = helpText.split(sc, 2);
-                    Integer countWhitespaces = helpText.countMatches(' ');
-
-                    if (parts.size() >= 2 && countWhitespaces <= 4) {
-                        if (String.isNotBlank(parts[0]) && String.isNotBlank(parts[1])) {
-                            return parts;
-                        }
+        /*******************************************************************************************************
+        * @description Retrieves an object and field api name from a DataImport__c field help text mapping.
+        *
+        * @param helpText Help text value of a DataImport__c field
+        * @return A list of strings containing an object api name and a field api name OR null if help text
+        * value doesn't contain a valid help text mapping.
+        */
+        private List<String> retrieveObjectAndFieldApiNames(String helpText) {
+            if (helpText.contains('.')) {
+                List<String> parts = helpText.split('\\.', 2);
```

Comment:
```suggestion
                List<String> parts = helpText.trim().split('\\.', 2);
```

@ptewson-sfdo does this seem sufficient to you with regards to clearing our preceding and/or trailing spaces?

Reply:
@psadie-sfdo - yep that looks great!